### DISCUSSION TITLE
Is it possible to define or translate the "all-day" label or the label from the "today" button?

### DISCUSSION THREAD

**Discussion by hrodrigues-belodigital** [GuavaCZ/calendar#32] [General]
Is it possible to define or translate the "all-day" label or the label from the "today" button?
![image](https://github.com/user-attachments/assets/5e2693f1-4be3-432f-9585-912ac663ff01)
![image](https://github.com/user-attachments/assets/143ac0d0-f6c3-4ddb-9307-8e61eef3be77)

**Comment by hrodrigues-belodigital** [comment]
`public function getOptions(): array
    {
        $this->setOption('allDayContent', 'Todo o Dia');
        $this->setOption('buttonText',['today' => 'Hoje']);
        return $this->evaluate($this->options);
    }`

Did this and it works but I don't like this solution because you can slightly see the "all-day" and "today" initial label before it switches to the ones I've defined. If anyone has a good solution I'm open for it.

**Reply by lukas-frey** [reply]
getOptions returns an array, just return allDayContent in the array and don't call setOption.

setOption sends an event and rerenders the calendar.

**Reply by hrodrigues-belodigital** [reply]
I figured it out later the mistake I was doing. Thank you for the help!

**Reply by lukas-frey** [reply]
You're welcome

**Reply by hrodrigues-belodigital** [reply]
I will use this discussion to also ask you if it is possible to switch between calendarViews using the setOption, or if there is any other way to do it.

**Reply by lukas-frey** [reply]
Any of the [options](https://github.com/vkurko/calendar#options) should be able to be changed by calling setOption.

![image](https://github.com/user-attachments/assets/7ab453f2-99ee-4c8f-9909-a89047f8c737)

**Reply by hrodrigues-belodigital** [reply]
Thank you.
